crates/typst-library/src/compute/construct.rs

```diff
@@ -754,6 +754,35 @@ pub fn range(
     Ok(array)
 }
 
+/// Loads a wasm plugin and return a plugin instance.
+///
+/// Plugins functions may accept strings as argument and
+/// return a single string.
+///
+/// ## Example { #example }
+/// ```example
+/// #let plugin = plugin("hello.wasm")
+///
+/// #plugin.hello() // returns the string hello world
+/// ```
+///
+/// Display: Plugin
+/// Category: data-loading
+#[func]
+pub fn plugin(
+    /// Path to a file.
+    path: Spanned<EcoString>,
+    /// The virtual machine.
+    vm: &mut Vm,
+) -> SourceResult<super::Value> {
```

Comment:
If you switch to `cast!` in `plugin.rs`, you can directly return `Plugin` here (and import it). The `#[func]` macro takes care of casting it to a `Value`. This way, the docs will also show the correct return type.